Fix flow file generation during compilation

houshuang opened the pull request:
I added two functions to frog-utils, values and entries, which are versions of Object.entries and Object.values which work properly with Flow (Jenny was complaining about this, and many on the Flow forum as well). 

While doing that, I realized that we've not been compiling Flow files after we changed the way we do initial_setup/watch! I fixed that. Otherwise, you do not get type definitions from external packages (for example when you import something from frog-utils in another package).

houshuang committed:
WIP

houshuang committed:
WIP

houshuang committed:
WIP

houshuang committed:
WIP

houshuang commented:
Looks like we have 150 Flow issues we haven't known about because Flow was not running properly :(

houshuang committed:
WIP

lfaucon committed:
Merge branch 'develop' into flowSilly

lfaucon committed:
WIP

lfaucon committed:
Merge branch 'develop' into flowSilly

lfaucon committed:
WIP

lfaucon committed:
WIP

lfaucon committed:
WIP

lfaucon committed:
WIP

lfaucon committed:
WIP

lfaucon committed:
Done

lfaucon committed:
Merge branch 'develop' into flowSilly

lfaucon committed:
WIP

lfaucon committed:
Deepscan

houshuang commented:
Wow, you were suddenly feeling inspired? Thanks a lot, that will make our work hopefully easier!

lfaucon commented:
Yeah, I started a bit before the weekend. We should test it and get it merged asap as it touches 71 files

lfaucon committed:
Eslint

houshuang commented:
Great, I'll start reviewing it. You have a tiny prettier error :)

On Mon, Apr 23, 2018 at 8:02 PM, Louis Faucon <notifications@github.com>
wrote:

> Yeah, I started a bit before the weekend. We should test it and get it
> merged asap as it touches 71 files
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chili-epfl/FROG/pull/960#issuecomment-383668092>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AADwh9FQa9Alpew3x_8dgI_RqWZMwY80ks5trhdMgaJpZM4TMybj>
> .
>



-- 
http://reganmian.net/blog -- Random Stuff that Matters

lfaucon committed:
Jest

lfaucon commented:
Go ahead, it's green :)

houshuang committed:
WIP

houshuang committed:
WIP

houshuang committed:
WIP

houshuang commented:
Good work! One important thing which I reverted by pushing to your branch, is that you added _id in {extra} in api/logs. First of all, we stopped adding ids, because we never use them, and sometimes we would get crashes where it somehow sent the same log message twice (for example because it was just reloading after sending) and crashed the server. Thus we just let mongo generate ids. 

Anyway, you were adding it to log extra, but remember that multiple logs will be sent with a single log extra, so this would guarantee multiple logs with the same _id!

I decided we might as well try to upgrade to the latest Flow version, which introduced some new errors, which I fixed. I also removed one file that is not in use anymore, and fixed a problem with the watchAll, after merging Prastut's updates there. 

I see you've done some changes on preview etc that are not just related to types - I cannot follow all the details, but I'm assuming you have tested extensively :)

houshuang reviewed (commented):
Great

houshuang commented:
Ah, I was wondering why I wasn't able to approve, but it's because I originally own this PR (since I started it) :). Anyway, since I pushed some more changes, I'll let you review them and merge when you're ready.

PS: We should really introduce dashboard specific types for the state and the dashboard data - the state means that mergeLog and prepareDataForDisplay are "in sync", and the data means that prepareDataForDisplay and Viewer are in sync - it also serves as useful documentation (I know I am often a bit confused about what exactly the Viewer expects, especially when it passes it on to a graph library etc).

houshuang committed:
WIP

houshuang commented:
I owe you a beer! :) And Romain should be very grateful to you :)

On Mon, Apr 23, 2018 at 10:57 PM, Louis Faucon <notifications@github.com>
wrote:

> Merged #960 <https://github.com/chili-epfl/FROG/pull/960>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chili-epfl/FROG/pull/960#event-1590149228>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AADwhxdBLJ4yZYOpWouAVxIRaXoxNAj_ks5trkApgaJpZM4TMybj>
> .
>



-- 
http://reganmian.net/blog -- Random Stuff that Matters
